### PR TITLE
Supplies layout

### PR DESCRIPTION
**Pull request by drodzewicz**
Creates Avatar component
Creates ShowMore component
Creates Supply Component
Creates SupplyUserTIle component
Creates ThreeDots component

### CODE REVIEW

**Comment by drodzewicz** [comment]
sorry for too many changes, forgot to change branches when working /testing other features :(

File: src/components/Sidebar/Sidebar.js [hunk 1]
```diff
@@ -3,26 +3,31 @@ import PropTypes from "prop-types";
...
-        <SideBarItem
-          key={navLink.navLink}
-          name={navLink.navLink}
-          icon={navLink.iconClass}
-        >
-          {navLink.navLink}
-        </SideBarItem>
-      ))}
+      {navlinks.map(navLink => {
+        const eventIdtemp = eventId !== undefined ? `/event/${eventId}` : "";
```
Comment (Nubzor) [inline src/components/Sidebar/Sidebar.js:10]: AFAICS eventId is not changing, you can do the check only once.

File: src/layout/EventTabs/Supply/Supply.js [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+import ShowMoreButton from "../../../components/ShowMoreButton/ShowMoreButton";
+import ThreeDots from "../../../components/ThreeDots/ThreeDots";
+import TextInput from "../../../components/Inputs/TextInput/TextInput";
+import Form from "../../../components/Form/Form";
+
+const Supply = ({ openModal }) => {
+  const setform = useContext(FormContext)[1];
+
+  const [supplyList, setsupply] = useState({
+    SupplyContainers: [
```
Comment (Nubzor) [inline src/layout/EventTabs/Supply/Supply.js]: I would move the data to a separate file.

File: src/layout/EventTabs/Supply/Supply.js [hunk 3]
```diff
@@ -0,0 +1,197 @@
+import React, { useState, useContext } from "react";
```
Comment (Nubzor) [inline src/layout/EventTabs/Supply/Supply.js:1]: This component is quite long it should be refactored.

File: package.json [hunk 4]
```diff
@@ -3,6 +3,7 @@
   "version": "0.1.0",
   "private": true,
   "dependencies": {
+    "axios": "^0.19.0",
```
Comment (Nubzor) [inline package.json]: why would you need axios?

File: src/layout/Profile/Profile.js [hunk 5]
```diff
@@ -53,11 +53,41 @@ const Profile = props => {
       </div>
       <div>
         <h2>My Events</h2>
-        <EventCard />
-        <EventCard />
-        <EventCard />
-        <EventCard />
-        <EventCard />
+        <EventCard
```
Comment (Nubzor) [inline src/layout/Profile/Profile.js:56]: wait, what? :) put the data to separate file and do a loop over it.

File: src/layout/OpeningPageLayout.js [hunk 6]
```diff
@@ -1,79 +1,136 @@
...
+  const registerForm = () => {
     return (
       <Form>
-        <TextInput placeholder="username" name="username" />
-        <TextInput placeholder="password" name="password" />
-        <TextInput placeholder="repeat password" name="password2" />
-        <TextInput placeholder="name" name="name" />
-        <TextInput placeholder="surname" name="surname" />
-        <TextInput placeholder="e-mail" name="email" />
+        <TextInput
```
Comment (Nubzor) [inline src/layout/OpeningPageLayout.js]: I would do it more generic, put the data into an array and then loop over it

File: src/layout/OpeningPageLayout.js [hunk 7]
```diff
@@ -1,79 +1,136 @@
...
 
-  openRegister = () => {
-    this.showModal();
-    this.setState({ chooseForm: false });
+  const handleRegsterChange = event => {
+    seNewState({
+      ...newState,
+      registerData: {
+        ...newState.registerData,
+        [`${event.target.name}`]: event.target.value
```
Comment (Nubzor) [inline src/layout/OpeningPageLayout.js:53]: smart 👍

File: src/components/Avatar/Avatar.js [hunk 8]
```diff
@@ -0,0 +1,7 @@
+import React from "react";
+
+const Avatar = ({ imageLink }) => {
+  return <img src={imageLink} alt="" />;
```
Comment (Nubzor) [inline src/components/Avatar/Avatar.js:4]: What about alt?

**Pull request merged**